src/restrictededitingexceptioncommand.js

```diff
@@ -38,10 +38,32 @@ export default class RestrictedEditingExceptionCommand extends Command {
 			const ranges = model.schema.getValidRanges( selection.getRanges(), 'restrictedEditingException' );
 
 			if ( selection.isCollapsed ) {
-				if ( valueToSet ) {
-					writer.setSelectionAttribute( 'restrictedEditingException', true );
-				} else {
+				const position = selection.getFirstPosition();
+
+				// When selection is inside a restricted text
+				if ( selection.hasAttribute( 'restrictedEditingException' ) ) {
```

Comment:
After all the changes this check is not required - basically, you need to check the `valueToSet` first (comment is somwhere below).